packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -11,8 +11,806 @@ import 'picker.dart';
+    return new CupertinoPicker(
+      scrollController: dayController,
+      offAxisFraction: offAxisFraction,
+      itemExtent: _kItemExtent,
+      useMagnifier: _kUseMagnifier,
+      magnification: _kMagnification,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          selectedDay = index + 1;
+          if (new DateTime(selectedYear, selectedMonth, selectedDay).day == selectedDay)
+            widget.onDateTimeChanged(new DateTime(selectedYear, selectedMonth, selectedDay));
+        });
+      },
+      children: new List<Widget>.generate(31, (int index) {
+        return childPositioning(new Text(localizations.datePickerDayOfMonth(index + 1)));
+      }),
+      looping: true,
+    );
+  }
+
+  Widget _buildMonthPicker(double offAxisFraction, Function childPositioning) {
+    return new CupertinoPicker(
+      scrollController: new FixedExtentScrollController(initialItem: selectedMonth - 1),
+      offAxisFraction: offAxisFraction,
+      itemExtent: _kItemExtent,
+      useMagnifier: _kUseMagnifier,
+      magnification: _kMagnification,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          selectedMonth = index + 1;
+          if (new DateTime(selectedYear, selectedMonth, selectedDay).day == selectedDay)
+            widget.onDateTimeChanged(new DateTime(selectedYear, selectedMonth, selectedDay));
+        });
+      },
+      children: new List<Widget>.generate(12, (int index) {
+        return childPositioning(new Text(localizations.datePickerMonth(index + 1)));
+      }),
+      looping: true,
+    );
+  }
+
+  Widget _buildYearPicker(double offAxisFraction, Function childPositioning) {
+    return new CupertinoPicker.builder(
+      scrollController: new FixedExtentScrollController(initialItem: selectedYear),
+      itemExtent: _kItemExtent,
+      offAxisFraction: offAxisFraction,
+      useMagnifier: _kUseMagnifier,
+      magnification: _kMagnification,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          selectedYear = index;
+          if (new DateTime(selectedYear, selectedMonth, selectedDay).day == selectedDay)
+            widget.onDateTimeChanged(new DateTime(selectedYear, selectedMonth, selectedDay));
+        });
+      },
+      itemBuilder: (BuildContext context, int index) {
+        if (index < widget.minimumYear)
+          return null;
+
+        if (widget.maximumYear != null && index > widget.maximumYear)
+          return null;
+
+        return childPositioning(new Text(localizations.datePickerYear(index)));
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Function> pickerBuilders = <Function>[];
+    List<double> columnWidths = <double>[];
+
+    if (localizations.datePickerDateOrder == DatePickerDateOrder.mdy) {
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values